packages/orbit-components/src/Modal/ModalSection/index.tsx

```diff
@@ -29,7 +29,7 @@ export const ModalSectionWrapper = ({
         "orbit-modal-section",
         "py-lg px-md lm:p-xl box-border w-full",
         "border-b-elevation-flat-border-color border-b border-solid",
-        "last-of-type:[&:not(:last-child)]:rounded-b-none",
+        "lm:last-of-type:rounded-modal last-of-type:[&:not(:last-child)]:rounded-b-none",
```

Comment:
So on large mobile, we have all the corners rounded, regardless of whether it's first- or last-of type like previously.

And then, if `!isMobileFullPage`, we ensure that the first-of-type on large mobile has rounded corners at the top, the last-of-type has rounded corners at the bottom.

I'm a bit confused 😅 there's some overlap between them, right?

Reply:
Thanks for the visual tests and screenshots, at least it looks like the behaviour should be correct now, I just don't have a good intuition for it when looking at the code.

Reply:
I had to revert this. The modal sections need to be handled with siblings selector because there might be an overlay div sibling for the close icon.

```
"[&_~_.orbit-modal-section:first-of-type]:rounded-t-none"
```